Sources/PublishCLICore/CLI.swift

```diff
@@ -42,8 +42,13 @@ public struct CLI {
             let deployer = WebsiteDeployer(folder: folder)
             try deployer.deploy()
         case "run":
-            let runner = WebsiteRunner(folder: folder)
-            try runner.run()
+            if arguments.count > 2, arguments[2].contains("-p=") || arguments[2].contains("--port="), let portNumber = extractPortNumber(from: arguments[2]) {
```

Comment:
This statement is quite hard to follow. Mind separating it into a dedicated method instead? I'm thinking the call site could look something like this:
```swift
let portNumber = extractPortNumber(from: arguments)
let runner = WebsiteRunner(folder: folder, portNumber: portNumber)
try runner.run()
```
That way the `extractPortNumber` method could contain all of this complexity, and you could use the `guard` statement to exit early, keeping the logic very neat and clean.

Reply:
I saw the "=" notation on command lines quite often, therefore I went with that.
Your approach reduces complexity, therefore I changed the "=" :). Also I caught myself multiple times writing "-p 3000" :D.
The complete logic of validating and returning the port number from the cli arguments was moved in the "extractPortNumber" method